Project/zepeto-modules/Assets/Zepeto Gesture Module/ZepetoScript/GestureLoader.ts

```diff
@@ -51,24 +55,66 @@ export default class GestureLoader extends ZepetoScriptBehaviour {
         newThumb.GetComponent<Thumbnail>().content = content;
 
         // Button Listener for each thumbnail
-        newThumb.GetComponent<Button>().onClick.AddListener(() => {
+        newThumb.GetComponent<Button>().onClick.AddListener(() => {  
             this.LoadAnimation(content);
         });
-        
         this.thumbnails.push(newThumb);
     }
 
     // 3. Loading Animation
-    private LoadAnimation(content: Content) {
+    private LoadAnimation(content: Content ) {
         // Verify animation load
         if (!content.IsDownloadedAnimation) {
             // If the animation has not been downloaded, download it.
             content.DownloadAnimation(() => {
                 // play animation clip
-                this._myCharacter.SetGesture(content.AnimationClip);
-            });
+                this.runAnimation(content.AnimationClip, content.Keywords)
+            });    
         } else {
-            this._myCharacter.SetGesture(content.AnimationClip);
+            this.runAnimation(content.AnimationClip, content.Keywords)
+        }
+    }
+       
+    // A function to run an animation, 
+    private runAnimation(animation: AnimationClip, type: OfficialContentType[] )
+    {        
+        //if there is another gesture running, stop the coroutine and cancel the gesture
+        if(this.gestureLoop)
+        {
+            this.StopCoroutine(this.gestureLoop)
+        }
+        this._myCharacter.CancelGesture()
+
+        // check if isLooping is true and it is not a pose
+        if(this.isLooping === true && this._checkIfGestureIsPose(type) === false)
```

Comment:
Can't we omit the `=== true`and `===false`?

Reply:
I thought writing like that will improve the code readability a little bit more. But I consider make a change.

Reply:
Yes, this may be about preference.
but since it is prominent that we'll be expecting both `A` and `B` to be evaluated to truthy or falsy in `if (A && B)`, I think `=== true(false)` is redundant.

Reply:
I got it. And the variables arr already boolean. Thanks for the input. Will make the change acccordingly.